doc/nrf/ug_bt_mesh_vendor_model_dev_overview.rst

```diff
@@ -0,0 +1,325 @@
+.. _ug_bt_mesh_vendor_model_dev_overview:
+
+Model development overview
+##########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+To implement a new Bluetooth mesh model, apply the steps described in this step-by-step model development process overview.
+
+Defining a model identifier
+***************************
+
+Models define basic functionality of a node in the Bluetooth mesh network.
+Once it is decided on what functionality the model will define, its identifier must be specified to be able to recognize it among the other models in the mesh network.
+
+As defined by the `Bluetooth Mesh Profile Specification`_, the vendor model identifiers are composed of two unique 16-bit values specifying a company identifier (Company ID), and a vendor-assigned model identifier (Model ID) tied to this Company ID.
```

Comment:
We stick to sentence casing in general, unless we are referencing the spec with its full name, including version and all (and then probably also linking directly to that specific version too).
-> Bluetooth mesh profile specification